### PR TITLE
Token providers can use the member address for token creation

### PR DESCRIPTION
**Pull request by yuce**

### CODE REVIEW

File: hazelcast/security.py [hunk 1]
```diff
@@ -1,13 +1,17 @@
 from hazelcast.six import string_types
+from hazelcast.core import Address
 
 
 class TokenProvider(object):
     """TokenProvider is a base class for token providers."""
 
-    def token(self):
-        # type: (TokenProvider) -> bytes
+    def token(self, address):
```
Comment (mdumandag) [inline hazelcast/security.py:8]: Does it make sense to make this optional by providing `None` as the default value? In basic use cases, you don't need an address, you can simply ignore it.

Comment (yuce) [reply hazelcast/security.py]: Yeah, that'd be better: https://github.com/hazelcast/hazelcast-python-client/pull/455/commits/57f3d488c0cc7b3959134576e29c97f6eddde35d

File: tests/unit/token_provider_test.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import unittest
 
 from hazelcast.security import BasicTokenProvider
+from hazelcast.core import Address
```
Comment (mdumandag) [inline tests/unit/token_provider_test.py:4]: Minor: Unused import

**Review by mdumandag — APPROVED** [review]

File: tests/unit/token_provider_test.py (hunk 2 above)
Comment (yuce) [reply tests/unit/token_provider_test.py]: Oops: https://github.com/hazelcast/hazelcast-python-client/pull/455/commits/d9ee08afd46e31f6cbbf8c9217a3e6ecddf418fc

**Comment by mdumandag** [comment]
verify

**Comment by yuce** [comment]
verify

**Pull request merged**